platform/fabricx/core/finality/listenermanager.go

```diff
@@ -64,21 +70,27 @@ func (p *listenerManagerProvider) NewManager(ctx context.Context, network, chann
 		return nil, err
 	}
 
-	lm, err := newNotifi(ctx, cfg)
+	lm, err := newNotifi(p.baseCtx, cfg)
 	if err != nil {
 		return nil, err
 	}
 
+	// cache the newly created instance
+	p.managers[key] = lm
+
 	// Start listening in background
 	// Listen() is a blocking method
 	go func() {
 		if err := lm.listen(); err != nil && !errors.Is(err, context.Canceled) {
 			logger.Errorf("Notification listener stream terminated unexpectedly: %s", err)
 		}
+		// Clean up the manager from the map when listen exits
+		p.managersMu.Lock()
+		delete(p.managers, key)
+		p.managersMu.Unlock()
```

Comment:
Maybe we should add a test so make sure that `p.managers` is eventually empty once we shutdown.